src/Microsoft.Azure.AppConfiguration.AspNetCore/AzureAppConfigurationRefreshMiddleware.cs

```diff
@@ -24,9 +25,12 @@ public AzureAppConfigurationRefreshMiddleware(RequestDelegate next, IConfigurati
 
         public async Task InvokeAsync(HttpContext context)
         {
-            foreach (var refresher in Refreshers)
+            using (var flowControl = ExecutionContext.SuppressFlow())
```

Comment:
I think it needs a comment to explain the reasoning. A suggestion
```
//
// Configuration refresh is meant to execute as an isolated background task.
// To prevent access of request-based resources, such as HttpContext, we detach the execution context within the refresh operation.
```

Reply:
Did you get a chance to iterate on this?